lib/__tests__/project.spec.js

```diff
@@ -0,0 +1,224 @@
+import { existsSync, unlinkSync } from 'node:fs';
+import { readFile } from 'node:fs/promises';
+import path from 'node:path';
+import { describe, it, expect, vi } from 'vitest';
+import nock from 'nock';
+import sharp from 'sharp';
+import { resolveFixturePath } from './helpers.js';
+import { parseProject } from '../project';
+
+vi.mock('sharp');
+
+const pkg = JSON.parse(await readFile('package.json'));
+
+describe('parseProject', () => {
+  describe('rejects', () => {
+    it('when dir not lowercase', () => {
+      const p = resolveFixturePath('a-Project');
+      expect.assertions(1);
+      return parseProject(p).catch((err) => {
+        expect(err.message)
+          .toBe('Directory name must be all lowercase and received a-Project');
+      });
+    });
+
+    it('when dir contains invalid chars', () => {
+      const p = resolveFixturePath('a-project #');
+      expect.assertions(1);
+      return parseProject(p).catch((err) => {
+        expect(err.message)
+          .toBe('Directory name must only contain alphanumeric characters and hyphens (-)');
+      });
+    });
+
+    it('when language not supported', () => {
+      const p = resolveFixturePath('01-project-bad-lang');
+      expect.assertions(1);
+      return parseProject(p, {
+        repo: 'Laboratoria/bootcamp',
+        version: '1.0.0',
+      }).catch((err) => {
+        expect(err.message).toBe('Unsupported lang: it');
+      });
+    });
+
+    it('when dir doesnt exist', () => {
+      expect.assertions(2);
+      return parseProject('01-foo').catch((err) => {
+        expect(err.message).toMatch(/no such file or directory/);
+        expect(err.code).toBe('ENOENT');
+      });
+    });
+
+    it('when README.md is empty', () => {
+      const p = resolveFixturePath('topic-empty');
+      expect.assertions(2);
+      return parseProject(p).catch((err) => {
+        expect(err.message).toBe(`${path.join(p, 'README.md')} is empty`);
+        expect(err.path).toBe(path.join(p, 'README.md'));
+      });
+    });
+
+    it('when README.md doesnt start with h1', () => {
+      const p = resolveFixturePath('01-a-project-without-a-title');
+      const p2 = resolveFixturePath('01-a-project-without-a-bad-title');
+      expect.assertions(4);
+      return parseProject(p)
+        .then(() => {
+          throw new Error('This should never happen');
```

Comment:
why is this `then` here? me parece como 2 pruebas compuesto en uno

Reply:
Originalmente el `then` estaba ahí para garantizar que la promesa no se resolvía, sino que se rechazaba, y antes no estaba `expect.assertions()`, con lo cual si la promesa se resolvía en vez de rechazarse, y no hacíamos nada en `then`, el test pasaría sin que llegue a ejecutarse el `catch`. Ahora que tenemos `expect.assertions(4);` antes de invocar la promesa, ya no necesitamos el `then` para garantizar que se haya ejecutado el `catch`. Voy a proceder a quitar el `then` :muscle:

Reply:
Ver https://github.com/Laboratoria/curriculum-parser/commit/26fbea9475d3443df537468178e3d117d02edd73